website/docs/extensions/write/index.md

```diff
@@ -0,0 +1,135 @@
+---
+sidebar_position: 1
+title: Writing
+description: Writing my first extension
+tags: [podman-desktop, extension, writing]
+keywords: [podman desktop, extension, writing]
+---
+
+# Writing your first extension
+
+Podman Desktop extensions rely on `JavaScript` or `TypeScript`.
+
+Extensions are self-contained, all runtime dependencies are inside the final binary.
+
+#### Scaffold the project
+
+1. Setup a Node.js project with `TypeScript`. Use for example `npm init` command.
+
+1. Bring the development dependency to Podman Desktop API. It allows to import the API and register the contributions.
+
+  Note that it is a dependency that brings only the definition of the API. It will not fetch any Podman Desktop internals.
+
+  ```json
+    "devDependencies": {
+      "@podman-desktop/api": "latest",
+    }
+  ```
+
+1. Setup the required metadata for the extension
+
+  ```json
+  {
+    "name": "my-extension",
+    "displayName": "My Hello World extension",
+    "description": "How to write my first extension",
+    "version": "0.0.1",
+    "icon": "icon.png",
+    "publisher": "benoitf",
+    "engines": {
+      "podman-desktop": "latest"
+    }
+  }
+  ```
+
+  In the `engines` section, describe the version of Podman Desktop that runs this extension.
+
+1. Include the main entry point of the extension
+
+  ```json
+    "main": "./dist/extension.js",
+  ```
+
+1. Add Hello World command contribution
+
+  ```json
+    "contributes": {
+      "commands": [
+        {
+          "command": "my.first.command",
+          "title": "My First Extension: Hello World"
+        }
+      ]
+    },
+  ```
+
+#### Writing the codebase of the extension
+
+The `package.json` file describes the metadata of the extension.
+Extension activates by launching the `activate` function of the main file.
+
+##### Entry point
+The extension entry point (`main` entry in `package.json` file) might expose two functions:
+
+One is for activation (`activate`), one for deactivation (`deactivate`). This one is optional.
+
+Signature of the function can be synchronous
+
+```typescript
+export function activate(): void
+```
+
+or asynchronous
+
+```typescript
+export async function activate(): Promise<void>
+```
```

Comment:
```suggestion
The signature of the function can be either:

* synchronous

  ```typescript
  export function activate(): void
  ```

* asynchronous

  ```typescript
  export async function activate(): Promise<void>
  ```
```

Reply:
looks like I can't apply your suggestion ?

Reply:
GH UI doesn't like suggestions with code blocks. I'll try in smaller chunks